### PR TITLE
Restore NVTT

### LINKED ISSUE #5866: Missing libpng12 for texture compression on Linux
Running tests on Linux I got DLLNotFoundException for nvtt (NVidia Texture Tools) because it depends on libpng 1.2. I had libpng installed, but not 1.2. Installing it on Arch Linux from the official repositories was easy and fixed the issue. We should probably include libpng 1.2 as a dependency.

cc @cra0zy

### LINKED ISSUE #6093: MGCB - NVidia.TextureTools Exception when processing a texture
Monogame `3.7.0.966`, then tried with `3.7.0.1126` and still the same problem.

I am randomly getting this exception when processing assets with MGCB:

```
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   in Nvidia.TextureTools.Compressor.nvttCompress(IntPtr compressor, IntPtr inputOptions, IntPtr compressionOptions, IntPtr outputOptions)
   in Microsoft.Xna.Framework.Content.Pipeline.Graphics.DxtBitmapContent.TryCopyFrom(BitmapContent sourceBitmap, Rectangle sourceRegion, Rectangle destinationRegion)
   in Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent.Copy(BitmapContent sourceBitmap, Rectangle sourceRegion, BitmapContent destinationBitmap, Rectangle destinationRegion)
   in Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent.Copy(BitmapContent sourceBitmap, BitmapContent destinationBitmap)
   in Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureContent.ConvertBitmapType(Type newBitmapType)
   in Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureProfile.ConvertTexture(ContentProcessorContext context, TextureContent content, TextureProcessorOutputFormat format, Boolean isSpriteFont)
```

It happens when processing textures of 3D models. what's worse is that it doesn't happens always.... for now the only solution is to try to recompile until it works. But it's a nightmare because we have our project in TeamCity, so we have to manually queue new rebuilds until it compiles.

### LINKED ISSUE #6145: System.DllNotFoundException: nvtt on MacOS 10.13
Downloaded MonoGame 3.6 from the site.
Downloaded MonoGame 3.6 PipeLine GUI tool for MacOS.

Whenever trying to build, MGCB.exe fails with DllNotFoundException: nvtt.

See full output of build:
```
Project "/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/GameTest.Desktop.csproj" (Build target(s)):

Target RunContentBuilder:
    /Library/Frameworks/Mono.framework/Versions/Current/bin/mono "/Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/Tools/MGCB.exe" /@:"/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/GameContent.mgcb" /platform:DesktopGL /outputDir:"/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/bin/DesktopGL" /intermediateDir:"/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/obj/DesktopGL" /quiet
    /Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/ship2.x
    EXEC : warning : Asset metal.jpg built multiple times with different settings.
    	/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/metal.jpg
    		Could not compress texture. Required shared lib is missing. System.DllNotFoundException: nvtt
      at (wrapper managed-to-native) Nvidia.TextureTools.Compressor:nvttCreateCompressor ()
      at Nvidia.TextureTools.Compressor..ctor () [0x00006] in <3f660b47cee74ab78456ee600c3e6d62>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.DxtBitmapContent.TryCopyFrom (Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent sourceBitmap, Microsoft.Xna.Framework.Rectangle sourceRegion, Microsoft.Xna.Framework.Rectangle destinationRegion) [0x00140] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent.Copy (Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent sourceBitmap, Microsoft.Xna.Framework.Rectangle sourceRegion, Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent destinationBitmap, Microsoft.Xna.Framework.Rectangle destinationRegion) [0x0011f] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent.Copy (Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent sourceBitmap, Microsoft.Xna.Framework.Content.Pipeline.Graphics.BitmapContent destinationBitmap) [0x0004c] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.GraphicsUtil.Compress (System.Type targetType, Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureContent content, System.Boolean generateMipMaps) [0x0007f] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.GraphicsUtil.CompressDxt (Microsoft.Xna.Framework.Graphics.GraphicsProfile profile, Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureContent content, System.Boolean generateMipMaps, System.Boolean isSpriteFont) [0x00063] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.DefaultTextureProfile.PlatformCompressTexture (Microsoft.Xna.Framework.Content.Pipeline.ContentProcessorContext context, Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureContent content, Microsoft.Xna.Framework.Content.Pipeline.Processors.TextureProcessorOutputFormat format, System.Boolean generateMipmaps, System.Boolean isSpriteFont) [0x00067] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureProfile.ConvertTexture (Microsoft.Xna.Framework.Content.Pipeline.ContentProcessorContext context, Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureContent content, Microsoft.Xna.Framework.Content.Pipeline.Processors.TextureProcessorOutputFormat format, System.Boolean generateMipmaps, System.Boolean isSpriteFont) [0x0003d] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
    /Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/ship2.x : error : Processor 'TextureProcessor' had unexpected failure!
    System.DllNotFoundException: nvtt
      at Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureProfile.ConvertTexture (Microsoft.Xna.Framework.Content.Pipeline.ContentProcessorContext context, Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureContent content, Microsoft.Xna.Framework.Content.Pipeline.Processors.TextureProcessorOutputFormat format, System.Boolean generateMipmaps, System.Boolean isSpriteFont) [0x00095] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.Processors.TextureProcessor.Process (Microsoft.Xna.Framework.Content.Pipeline.Graphics.TextureContent input, Microsoft.Xna.Framework.Content.Pipeline.ContentProcessorContext context) [0x002bb] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at Microsoft.Xna.Framework.Content.Pipeline.ContentProcessor`2[TInput,TOutput].Microsoft.Xna.Framework.Content.Pipeline.IContentProcessor.Process (System.Object input, Microsoft.Xna.Framework.Content.Pipeline.ContentProcessorContext context) [0x00038] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
      at MonoGame.Framework.Content.Pipeline.Builder.PipelineManager.ProcessContent (MonoGame.Framework.Content.Pipeline.Builder.PipelineBuildEvent pipelineEvent) [0x00163] in <a08ce79bc40d4146bf2ee59422c41dea>:0 
    /Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/metal.jpg
    
    Unhandled Exception:
    System.DllNotFoundException: nvtt
      at (wrapper managed-to-native) Nvidia.TextureTools.Compressor:nvttDestroyCompressor (intptr)
      at Nvidia.TextureTools.Compressor.Finalize () [0x00000] in <3f660b47cee74ab78456ee600c3e6d62>:0 
    [ERROR] FATAL UNHANDLED EXCEPTION: System.DllNotFoundException: nvtt
      at (wrapper managed-to-native) Nvidia.TextureTools.Compressor:nvttDestroyCompressor (intptr)
      at Nvidia.TextureTools.Compressor.Finalize () [0x00000] in <3f660b47cee74ab78456ee600c3e6d62>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/MonoGame.Content.Builder.targets(90,5): error MSB3073: The command "/Library/Frameworks/Mono.framework/Versions/Current/bin/mono "/Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/Tools/MGCB.exe" /@:"/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/GameContent.mgcb" /platform:DesktopGL /outputDir:"/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/bin/DesktopGL" /intermediateDir:"/Users/CrazyLegz/Projects/GameTest/GameTest.Desktop/Content/obj/DesktopGL" /quiet" exited with code 255.
Done building target "RunContentBuilder" in project "GameTest.Desktop.csproj" -- FAILED.
```
When running from terminal with $MONO_LOG_LEVEL=debug there are some interesting entries, see the last item:
```
Mono: DllImport error loading library 'libnvtt.so': 'dlopen(libnvtt.so, 9): image not found'.
Mono: DllImport error loading library 'libnvtt.bundle': 'dlopen(libnvtt.bundle, 9): image not found'.
Mono: DllImport error loading library 'libnvtt': 'dlopen(libnvtt, 9): image not found'.
Mono: DllImport unable to load library 'dlopen(libnvtt, 9): image not found'.
Mono: DllImport attempting to load: 'nvtt'.
Mono: DllImport error loading library '/Library/Frameworks/MonoGame.framework/v3.6.0.1625/Tools/libnvtt': 'dlopen(/Library/Frameworks/MonoGame.framework/v3.6.0.1625/Tools/libnvtt, 9): image not found'.
Mono: DllImport error loading library '/Library/Frameworks/MonoGame.framework/v3.6.0.1625/Tools/libnvtt.dylib': 'dlopen(/Library/Frameworks/MonoGame.framework/v3.6.0.1625/Tools/libnvtt.dylib, 9): no suitable image found.  Did find:
	/Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/Tools//libnvtt.dylib: mach-o, but wrong architecture
```

### PR DESCRIPTION
**Pull request by tomspilman**
This PR restores NVTT for Windows, Mac, and Linux BC/DXT texture compression.

The solution in #6102 while better for ease of integration and fewer dependencies is way too slow for real world texture processing work.

To improve upon our previous NVTT implementation we've:

 - [Made a fork of NVTT](https://github.com/MonoGame/nvidia-texture-tools).
 - Updated to the latest NVTT.
 - Setup the build server to [build NVTT correctly for us](http://teamcity.monogame.net/project.html?projectId=Dependencies&guest=1).
 - Fixed NVTT to [link against the static VC++ runtime](https://github.com/MonoGame/nvidia-texture-tools/commit/5a4bface9c2d7afb8348d7d4fc2e2ac66288ee19).

With these changes it should be easy to update NVTT in the future and this finally fixes a few issues:

Fixes #6093
Fixed #6145

### CODE REVIEW

File: MonoGame.Framework.Content.Pipeline/Graphics/DxtBitmapContent.cs [hunk 1]
```diff
@@ -4,9 +4,90 @@
...
+            _content = content;
+
+            WriteData = new OutputOptions.WriteDataDelegate(WriteDataInternal);
+            BeginImage = new OutputOptions.ImageDelegate(BeginImageInternal);
+
+            // Keep the delegate from being re-located or collected by the garbage collector.
+            delegateHandleBeginImage = GCHandle.Alloc(BeginImage);
+            delegateHandleWriteData = GCHandle.Alloc(WriteData);
+
+            outputOptions.SetOutputHandler(BeginImage, WriteData);
```
[the remark this answers is not in the document]
Comment (tomspilman) [reply MonoGame.Framework.Content.Pipeline/Graphics/DxtBitmapContent.cs]: I am aware... i reworked the code last night.

Just haven't pushed it as i want to verify it is all working correctly before i do so.

**Comment by tomspilman** [comment]
@dellis1972  - https://github.com/MonoGame/MonoGame.Dependencies/pull/121 needs a merge then i can get this PR updated so it will all build.

**Comment by tomspilman** [comment]
I'm going to rebase after https://github.com/MonoGame/MonoGame/pull/6252 is merged so i can be sure these NVTT changes don't break the pipeline on Mac.

**Comment by tomspilman** [comment]
Woot... so the current unit tests all pass on Mac and Windows.  Will write a few more tests specifically for `DxtBitmapContent` to be sure it works.

**Comment by tomspilman** [comment]
So ran into this https://github.com/MonoGame/MonoGame/issues/6259 which i'm going to push off for another PR.

I added a few more tests which cover a few cases we have missed.  Still we need to decide on the right path to beef up our texture processing tests.  Comparing single colors isn't good enough to catch bugs and fuzzy image comparison is hard to do well.

**Comment by tomspilman** [comment]
@cra0zy @dellis1972 - on #5866 this is just a matter of fixing our Linux build correct?  It doesn't affect Mac?

**Comment by tomspilman** [comment]
FYI... going to push #5866 until a later PR once we have the new Linux box running and can build NVTT for Linux properly.

**Comment by tomspilman** [comment]
@cra0zy @Jjagg @dellis1972 - I think this PR is good to go.

Have been using it for the last few days in the office across a few different PCs and haven't had any issues with it.

**Comment by mrhelmut** [comment]
Glad to see this ready, our textures baking went from a few seconds to more than a minute when NVTT was removed.

**Comment by dellis1972** [comment]
Mac Build seems to work ok.

**Comment by cra0zy** [comment]
Merged!!!

**Pull request merged**